### PR TITLE
Add fake-source-cluster for testing the sync service

### PR DESCRIPTION
**Pull request by saza-ku**
#### What type of PR is this?

/area simulator
/kind feature

#### What this PR does / why we need it:

This PR adds a kwok cluster that will be the source cluster of import/sync when debugging.
This cluster starts by running `make docker_debug`.

#### Which issue(s) this PR fixes:

Part of #327 

#### Special notes for your reviewer:

When you edit `simulator/config.yaml` to enable `externalImportEnabled` and run `make docker_debug`, you will see `node-0`, `node-1` and `node-2` in the simulator UI.

/label tide/merge-method-squash

### CODE REVIEW

File: docker-compose-local.yml [hunk 1]
```diff
@@ -49,6 +53,38 @@ services:
       - KWOK_KUBE_APISERVER_PORT=3131
     networks:
       - simulator-internal-network
+  debug-real-cluster:
```
[the remark this answers is not in the document]
Comment (ordovicia) [reply docker-compose-local.yml:56]: I agree that "fake" is more appropriate than "debug".
"debug" seems to overly narrow the intended purpose of the cluster.

However, "fake" and "real" words are contradictory.
How about "fake-source-cluster"?

File: docker-compose-local.yml [hunk 2]
```diff
@@ -49,6 +53,38 @@ services:
...
+    ports:
+      - "3132:3132"
+    environment:
+      - KWOK_KUBE_APISERVER_PORT=3132
+    networks:
+      - simulator-internal-network
+    profiles:
+      - debug
+  init-debug-real-cluster:
+    image: bitnami/kubectl:1.30.2
```
Comment (ordovicia) [inline docker-compose-local.yml:74]: nit: I guess we can use `registry.k8s.io/kubectl`

File: Makefile [hunk 3]
```diff
@@ -68,10 +68,18 @@ docker_up_local:
 .PHONY: docker_build_and_up
 docker_build_and_up: docker_build docker_up_local
 
+.PHONY: docker_debug
```
[the remark this answers is not in the document]
Comment (saza-ku) [reply Makefile]: We can specify the profiles by an environment variable of `COMPOSE_PROFILES`, like this.

```
make docker_up_local -e COMPOSE_PROFILES=externalImportEnabled
```

So this PR doesn't fix Makefile.

**Comment by ordovicia** [comment]
Could you please also update `simulator/docs/running-simulator.md` to describe how to run the fake source cluster?

**Comment by saza-ku** [comment]
↑ Updated!

**Review by ordovicia — APPROVED** [review]
👍

File: simulator/docs/running-simulator.md [hunk 4]
```diff
@@ -23,3 +23,11 @@ make docker_build_and_up
 Then, you can access the simulator with http://localhost:3000.
 If you want to deploy the simulator on a remote server and access it via a specific IP (e.g: like http://10.0.0.1:3000/),
 please make sure that you have executed `export SIMULATOR_EXTERNAL_IP=your.server.ip` before running `docker compose up -d`.
+
+You can run a kwok cluster that acts as a fake source cluster of syncing the resources of a real cluster.
+This is helpful to debug the syncing service.
+You can do it by the following command.
```
Comment (sanposhiho) [inline simulator/docs/running-simulator.md:29]:
```suggestion
Additionally, you can run a kwok cluster that acts as a fake source cluster to try out [the resource importing feature](./import-cluster-resources.md).
```

File: docker-compose-local.yml [hunk 5]
```diff
@@ -19,12 +19,19 @@ services:
       - PORT=1212
       - KUBE_SCHEDULER_SIMULATOR_ETCD_URL=http://simulator-cluster:2379
       - KUBE_APISERVER_URL=http://simulator-cluster:3131
+    volumes:
+      - ./simulator/config.yaml:/config.yaml
```
Comment (sanposhiho) [inline docker-compose-local.yml:23]: Given we modify `Dockerfile` itself, please add the same fix for `/config.yaml` to docker-compose.yaml too.

File: docker-compose.yml [hunk 6]
```diff
@@ -3,6 +3,9 @@ services:
   simulator-server:
     image: registry.k8s.io/scheduler-simulator/simulator-backend:v0.2.0
     container_name: simulator-server
+    volumes:
+      - ./simulator/config.yaml:/config.yaml
+      - ./simulator/kubeconfig.yaml:/kubeconfig.yaml
```
Comment (sanposhiho) [inline docker-compose.yml:8]: We shouldn't mount this kubeconfig since we don't have fake-source-cluster in this case.

**Review by sanposhiho — APPROVED** [review]
/lgtm
/approve

**Pull request merged**